### PR TITLE
Expanding self-hosted M1 build

### PR DESCRIPTION
**Pull request by csomor**

### CODE REVIEW

File: .github/workflows/MacS-M1-make.yml [hunk 1]
```diff
@@ -22,13 +22,20 @@ jobs:
...
+        - build: release
+          config: --without-liblzma -with-zlib=builtin --with-osx_cocoa
 
     steps:
     - uses: actions/checkout@v2
       with:
         submodules: recursive
     - name: configure
-      run: mkdir -p $PWD/localbin_${{ matrix.arch }}; ./configure --prefix=$PWD/localbin_${{ matrix.arch }}
+      run: mkdir -p $PWD/localbin_${{ matrix.arch }}; ./configure --{{ matrix.config }} --with-cxx={{ matrix.cxx }} --prefix=$PWD/localbin_${{ matrix.arch }}
```
Comment (vadz) [inline .github/workflows/MacS-M1-make.yml:38]:
```suggestion
      run: mkdir -p $PWD/localbin_${{ matrix.arch }}; ./configure ${{ matrix.config }} --with-cxx=${{ matrix.cxx }} --prefix=$PWD/localbin_${{ matrix.arch }}
```

File: .github/workflows/MacS-M1-make.yml [hunk 2]
```diff
@@ -22,13 +22,20 @@ jobs:
       fail-fast: false
       matrix:
         arch: [arm64e, x86_64]
+        build: [debug, release]
+        cxx: [11, 17]
+        include:
+        - build: debug
+          config: --without-liblzma --enable-debug --with-zlib=builtin --with-osx_cocoa
```
Comment (vadz) [inline .github/workflows/MacS-M1-make.yml:29]: I'd use (just) `--disable-sys-libs`.

**Comment by vadz** [comment]
Stefan, I'm going to push a change to avoid GitHub CI rebuilds too for now, but in fact, I hope that once this works, we could put it directly into `ci.yml` -- or do we really need to have 2 different CI workflows?

File: .github/workflows/MacS-M1-make.yml (hunk 1 above)
Comment (csomor) [reply .github/workflows/MacS-M1-make.yml]: thanks

File: .github/workflows/MacS-M1-make.yml (hunk 2 above)
Comment (csomor) [reply .github/workflows/MacS-M1-make.yml]: ok, thanks

**Comment by csomor** [comment]
> Stefan, I'm going to push a change to avoid GitHub CI rebuilds too for now, but in fact, I hope that once this works, we could put it directly into `ci.yml` -- or do we really need to have 2 different CI workflows?

I wanted to target things by properly having ignore-paths, is this possible for two different situations mac and gtk ?

**Comment by vadz** [comment]
> 
> 
> > Stefan, I'm going to push a change to avoid GitHub CI rebuilds too for now, but in fact, I hope that once this works, we could put it directly into `ci.yml` -- or do we really need to have 2 different CI workflows?
> 
> I wanted to target things by properly having ignore-paths, is this possible for two different situations mac and gtk ?

Hmm, good point. This is indeed not currently supported, although I'll keep on eye on [this topic](https://github.community/t/path-filtering-for-jobs-and-steps/16447/14) to see if it changes in the future. and there are [some](https://github.com/dorny/paths-filter) [workaround](https://github.com/MarceloPrado/has-changed-path) already available.

**Comment by vadz** [comment]
Stefan, do you have any idea why all the PR builds now fails for x86_64? This is a bit annoying, as they're never green any longer and you have to go and check that it was really just this job that failed. Could it perhaps be disabled for now in master if it's not going to be fixed soon?

See e.g. https://github.com/wxWidgets/wxWidgets/actions/runs/769819393

**Comment by vadz** [comment]
Stefan, do you plan to return to this? It would be nice to have it in place, even in the most minimal form, before the next release, so that we could check that nothing gets broken at the last moment (as is tradition).

**Comment by csomor** [comment]
> Stefan, do you plan to return to this? It would be nice to have it in place, even in the most minimal form, before the next release, so that we could check that nothing gets broken at the last moment (as is tradition).

yes, definitely, I've updated the machine, I'll look at this in the next weeks
(in the meantime I also found out about the [skip ci] options in a commit message, seems useful ...)

**Comment by csomor** [comment]
@vadz do you have an idea how to solve this failing http test ?

https://github.com/csomor/wxWidgets/runs/3376182330?check_suite_focus=true

**Comment by vadz** [comment]
> 
> 
> @vadz do you have an idea how to solve this failing http test ?
> 
> https://github.com/csomor/wxWidgets/runs/3376182330?check_suite_focus=true

It's not the HTTP test which is failing, but `DateTimeTestCase`, the log of the HTTP stuff at the end is just informational.

This looks like a real bug that I should debug, but for now it might be enough to run the test with `TZ=UTC` or something like this.

**Comment by csomor** [comment]
@vadz thanks a lot that helped

things are looking better now, I had a problem once because I've added a second runner for my cloned csomor/wxWigets , to test my PRs, and I didn't discover where to change the http port used in the test. I guess I'll have to add a env var in the second runner.

**Comment by csomor** [comment]
@vadz I think this is in a useful state now. I don't know how to best proceed with the merge, I fear I'd add the travis file back ...

**Comment by vadz** [comment]
Thanks! I can merge this if you want.

**Comment by csomor** [comment]
> Thanks! I can merge this if you want.

this would be great, I fear causing havoc ... thanks a lot!

**Comment by vadz** [comment]
(Squash) merged now, thanks again for making this work!

**Comment by vadz** [comment]
@csomor Stefan, looking at this again, it's great that individual builds are so fast (<5 minutes!), but 8 of them still take ~40 minutes, which is still faster than AppVeyor but still takes some time. Do we really need all 4 builds? I think that it would be enough to have fewer builds if they covered all the values of arch/debug/cxx, e.g. maybe just arm64e-release-cxx11 and x86_64-debug-cxx17?

OTOH I think it would be very useful to enable ASAN for the tests under Mac too, if we can do this.

And I think it would be also useful to have a universal build, e.g. one using `--enable-universal_binary=arm64,x86_64 --disable-shared --disable-debug` (this is what I use for my own binaries).

What do you think?

**Comment by csomor** [comment]
> @csomor Stefan, looking at this again, it's great that individual builds are so fast (<5 minutes!), but 8 of them still take ~40 minutes, which is still faster than AppVeyor but still takes some time. Do we really need all 4 builds? I think that it would be enough to have fewer builds if they covered all the values of arch/debug/cxx, e.g. maybe just arm64e-release-cxx11 and x86_64-debug-cxx17?
> 
> OTOH I think it would be very useful to enable ASAN for the tests under Mac too, if we can do this.
> 
> And I think it would be also useful to have a universal build, e.g. one using `--enable-universal_binary=arm64,x86_64 --disable-shared --disable-debug` (this is what I use for my own binaries).
> 
> What do you think?

absolutely, I thought we might have some checkerboard selection of the matrix, so that we touch every config, but not all possible combinations of every option ...

**Comment by vadz** [comment]
> absolutely, I thought we might have some checkerboard selection of the matrix, so that we touch every config, but not all possible combinations of every option ...

Yes, exactly.

Would you be fine with the builds mentioned above, i.e. arm64e-release-cxx11, x86_64-debug-cxx17 and static univ-release-cxxwhatever (what would you prefer to use for this one?)?

**Comment by csomor** [comment]
> > absolutely, I thought we might have some checkerboard selection of the matrix, so that we touch every config, but not all possible combinations of every option ...
> 
> Yes, exactly.
> 
> Would you be fine with the builds mentioned above, i.e. arm64e-release-cxx11, x86_64-debug-cxx17 and static univ-release-cxxwhatever (what would you prefer to use for this one?)?

I haven't thought about how to implement the checkerboard, that's why I also left out cxx14 - not to increase the combinations even more - but if we can select then I'd like cxx14 for that

**Pull request closed without merging**